src/vm/ceeload.cpp

```diff
@@ -6461,6 +6461,36 @@ void Module::NotifyDebuggerUnload(AppDomain *pDomain)
 }
 
 #if !defined(CROSSGEN_COMPILE)
+using GetTokenForVTableEntry_t = mdToken(STDMETHODCALLTYPE*)(HMODULE module, BYTE**ppVTEntry);
+
+
+HMODULE GetIJWHostModule()
+{
+#if !defined(FEATURE_PAL)
+    static HMODULE ijwHostModule = nullptr;
+    if (ijwHostModule == nullptr)
+    {
+        ijwHostModule = CLRGetModuleHandle(W("ijwhost.dll"));
+    }
+
+    return ijwHostModule;
+#else
+    return nullptr;
+#endif // FEATURE_PAL
+}
+
+GetTokenForVTableEntry_t GetTokenGetterFromHostModule()
```

Comment:
Why do we need this callback when .NET Framework worked fine without it?

Reply:
.NET Framework did all three of these functions (`GetTokenForVTableEntry`, `GetTargetForVTableEntry`, `SetTargetForVTableEntry`) as callbacks to `mscoree.dll`. All three live in the mscoree source.

The reason we need to repeatably check if ijwhost was loaded is because unlike mscoree.dll, ijwhost.dll will ship with the application and may not always be available. Additionally, it allows us to only go to the ijwhost shim when something that uses ijwhost is loaded.

Reply:
Is it guaranteed that there is going to be just one ijwhost.dll in the application? I see that you are doing stuff with assembly load contexts...

Reply:
`ijwhost.dll` will be statically linked to the IJW assembly via the native linker. As far as I understand the Windows loader, if two libraries that each natively statically reference a native dll named `ijwhost.dll` with no path, then after one `ijwhost.dll` is loaded, then later libraries will use the same `ijwhost.dll` module that has been loaded into memory. This loading should not be interfered with because of ALCs AFAIK.

Reply:
ok